src/weatherAnalysis.java

```diff
@@ -0,0 +1,48 @@
+import java.util.*;
+
+public class weatherAnalysis extends Averager{
+    public static void main(String[] args) {
+        int[] tempArray = intergerify(args);
+        analyzer(tempArray);
+    }
+
+    public static String analyzer(int[] arg){
+        HashSet<Integer> temperatures = new HashSet<>();
+        HashSet<Integer> allTemps = new HashSet<>();
+        int max = -500;
+        int min = 500;
+        for(int currentVal : arg){
+            if(currentVal>max){
+                max = currentVal;
+            }
+            if(currentVal<min){
+                min = currentVal;
+            }
+            temperatures.add(currentVal);
+        }
+        for(int currentTemp = min;currentTemp<max; currentTemp++){
+            allTemps.add(currentTemp);
+        }
+
```

Comment:
It is preferred to use the `for-each` loop instead of explicitly creating an `Iterator`; the `for-each` will handle iteration for you automatically:
```
for (int temp : temperatures) {
    // do stuff          
}
```